### DISCUSSION TITLE
How to setup a single active consumer in a stream

### DISCUSSION THREAD

**Discussion by heldergomes** [rabbitmq/rabbitmq-server#10919] [Questions]
Hello folks,

I working in a project which we have a really nice problem that we could solve with stream, because we have many consumers that need to read the same message, however because each consumer has many pods in k8s, we need to use single active consumer.

We are struggling to figure out how to set up this argument, it seems that the stream doesn't have the option to put single active consumer in the rabbit interface.

Could someone give a clue how I can setup the single active consumer feature in my stream ?

**Comment by lukebakken** [comment]
One of my co-workers who knows more about this will probably chime in (paging @acogoluegnes and @Gsantomaggio), but I thought I'd first request some information from you:
* RabbitMQ and Erlang version
* RabbitMQ streams client library you're using, and version
* Ideally, a runnable code sample showing what you have tried

I'm assuming you have read the docs and blog post:
- https://www.rabbitmq.com/docs/streams#single-active-consumer
- https://www.rabbitmq.com/blog/2022/07/05/rabbitmq-3-11-feature-preview-single-active-consumer-for-streams

**Comment by michaelklishin** [comment]
[There is a blog post on that](https://www.rabbitmq.com/blog/2022/07/05/rabbitmq-3-11-feature-preview-single-active-consumer-for-streams).

**Reply by michaelklishin** [reply]
I'd agree that the example should be moved to the Streams guide but it does link to the blog post already.

**Comment by heldergomes** [comment]
I've read those posts, but because our applications is written in Python, is not that clear, anyways, I will try to test this feature with jave in order to figure it out.

**Reply by Gsantomaggio** [reply]
@heldergomes   here and example with the python stream client  https://github.com/qweeze/rstream?tab=readme-ov-file#single-active-consumer
